src/main/java/at/hannibal2/skyhanni/features/misc/TabWidgetSettings.kt

```diff
@@ -0,0 +1,104 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils.getInventoryName
+import at.hannibal2.skyhanni.utils.InventoryUtils.getUpperItems
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraft.inventory.Slot
+import net.minecraft.item.Item
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class TabWidgetSettings {
+
+//     private val mainPageWidgetPattern = "^§7Currently:.*".toRegex()
+//
+//     private val subPageWidgetPattern = "^§eClick to .*".toRegex(RegexOption.IGNORE_CASE)
+//
+//     private val shownSettingPattern = "Shown .* Setting.* |.*Widget Settings".toRegex(RegexOption.IGNORE_CASE)
+
+    private val repoGroup = RepoPattern.group("tab.widget.setting")
+    private val mainPageSettingPattern by repoGroup.pattern("gui","^WIDGETS.*")
+    private val mainPageWidgetPattern by repoGroup.pattern("main","^§7Currently:.*")
+    private val subPageWidgetPattern by repoGroup.pattern("sub","^§eClick to .*")
+    private val shownSettingPattern by repoGroup.pattern("show","Shown .* Setting.* |.*Widget Settings")
```

Comment:
the group should be named patternGroup and all patterns should be split over 4 lines each. Also there is no need to say where the pattern starts with ^ because pattern.matchMatcher matches the whole string already